### ISSUE TITLE
proposal: runVault to extend runDefault

### ISSUE THREAD

**Issue by psychobolt** [dotenvx/dotenvx#132]
Hi, great library. I was wondering if it would be possible to extend RunDefault e.g.

```js
class RunVault extends RunDefault {
// super...
}
```

That would allow to overload also variables from multiple environment files e.g.

```sh
DOTENV_KEY=<key> dotenvx run -f ./.env.defaults -f ./.env -e "HELLO=world" --overload
```

I've tested this change in my personal project and seems to work well. See my [patch](https://github.com/psychobolt/vite-storybook-boilerplate/pull/570/files#diff-74d699825decccbb21ac85bfdec23a08993ee2f8b155930cf69ea41adc051573).

**Comment by motdotla** [comment]
interesting, yes, highly considering this. 

can you tell me more about the context of your use case though? why do you need this versus just using the values in .env.vault. is this typically in production or in development that you are using this approach?

**Comment by psychobolt** [comment]
When configuring environment variables with the CI provider, e.g. bitbucket cloud, we may want to store private variables in the vault only for test e.g. CHROMATIC_PROJECT_TOKEN. So if we set DOTENV_KEY globally, we have to ensure to unset it each time we run `dotenvx` or else it will not include environment variables from file with our other commands. The reason is we have certain environments saved in a `.env.defaults` file that is shared with our team project workspace without loading the vault.

**Comment by motdotla** [comment]
thanks, this is very helpful.

**Comment by motdotla** [comment]
working on this next

**Comment by motdotla** [comment]
to do this correctly (the least developer surprise), it should respect the order the flags are placed in - no matter if it is a mix of flags (`--env-file`, `--env-vault-file`, `--env`). A dev should be able to compose those in any order they want and be read left from right. If overload passed than the furthest right 'wins'.

give me some time on this. have to rejig some things and add a mechanism to track ordering of different flags.

**Comment by motdotla** [comment]
almost completed. this actually unlocks the possibility for multiple `.env.vault` files - though currently I can't think of a good use case for that. maybe migrating encryption across very large deployments where you can guarantee things being in sync.

**Comment by motdotla** [comment]
available in `v0.29.0` (publishing now)

**Comment by motdotla** [comment]
wait, darn. i think one bug - particularly with your main scenario. adding a test and cleaning up and will release patch.

**Comment by motdotla** [comment]
patched `v0.29.1`

**Comment by psychobolt** [comment]
Thanks, I already see that my tests are passing with the new version.
